### ISSUE TITLE
In Kolibri deployments, multiple frontend instances cause duplicate processing of tasks

### ISSUE THREAD

**Issue by bjester** [learningequality/kolibri#9641]
## Observed behavior

In Kolibri cloud deployments, such as Kolibri Data Portal (KDP), we start multiple instances of Kolibri, each within an individual Kubernetes pod. Starting several instances of Kolibri means we also start up separate task workers within each pod. Doing so creates the opportunity for the duplication of task processing, since Kolibri's task processing does not protect against this happening.

## Errors and logs
In KDP, we receive errors similar to the following, which are raised in the asynchronous processing of Kolibri syncs:
```
django.db.utils.IntegrityError: duplicate key value violates unique constraint "morango_databasemaxcounter_instance_id_partition_99e4f1fb_uniq"
```
which leads to
```
kolibri_data_portal.management.commands.sync_proceed_to.SyncProceedToError: Failed to finalize <transfer session ID>
```
This would appear to be an issue, but isn't because when you look at the Morango `TransferSession` referenced by the error, you can see that it did complete successfully and didn't actually "fail to finalize". In addition, you can also see that the transfer was completed prior to the `duplicate key value violates unique constraint` error, which indicates that there was another worker trying to process the same task, hence the reason for `duplicate key` error.

## Expected behavior

Kolibri should implement a distributed locking mechanism which prevents Kolibri task workers from picking up tasks which have already been picked up and "locked" by a worker. It may be useful to annotate which worker picked up the task on the job record, and have workers filter for tasks using that; for instance the combination of worker hostname and process ID should work appropriately in deployments like KDP and would be useful for debugging too. 

## User-facing consequences

This can cause significant database bottlenecks on KDP's side since these tasks can retry their operations when they encounter transaction serialization errors due to their more aggressive transaction isolation. Also, the flood of these errors obscure real errors that need to be resolved.

## Context

Kolibri 0.14+0.15
Linux + Google Cloud

**Comment by bjester** [comment]
@rtibbles

**Comment by rtibbles** [comment]
Noting that in the fix above, I have not added the process annotation. I think that's feasible, but would be better done in 0.16, as it will require schema changes to add the column, so when the above is reviewed and merged, I'll make a follow up issue for that.

**Comment by rtibbles** [comment]
Fixed in #9684 - follow up added in #9710.
